### PR TITLE
CRUD: Java examples

### PR DESCRIPTION
**Pull request by anezthes**
Some consistency issues across Fusion/Flow:

### Java
- Column headers use title case (default) ()
- Setting `setAsRequiredEnabled` on the binder crashes the example. Used `setRequiredIndicatorVisible` instead but fields never become invalid. (All fields are required by default in Fusion.)

### TypeScript
- Missing column headers (default)
- Filter label lack whitespace, e.g. "firstname"

This paragraph is perhaps a bit misleading as the example only shows how to remove existing columns. Adding and/or configuring columns isn't possible without replacing the default grid.

### Java
- Can't bind birthday to binder (`DatePicker` uses `LocalDate`, dataset uses `Date`, changing the dataset crashed the examples).

### TypeScript
- First example is missing filters

### CODE REVIEW

**Review by magi42 — APPROVED** [review]
The AsciiDoc looks OK, although I couldn't see the preview as it had failed for some reason. Didn't review TS code.

File: articles/ds/components/crud/index.asciidoc [hunk 1]
```diff
@@ -229,13 +259,25 @@ For more information about sorting and filtering, please see the basic <<../grid
...
+
+...
 
-// [source,java]
-// ----
-// include::../../../../src/main/java/com/vaadin/demo/component/crud/CrudSortingFiltering.java[render,tags=snippet,indent=0,group=Java]
-// ----
+include::{root}/src/main/java/com/vaadin/demo/component/crud/CrudSortingFiltering.java[render,tags=snippet2,indent=0,group=Java]
+
+...
```
[the remark this answers is not in the document]
Comment (jouni) [reply articles/ds/components/crud/index.asciidoc:273]: The ellipsis is shown in the collapsed source code view, indicating that a part of the code is hidden from between the two snippets before and after it. The ellipsis won't be visible when the source code is expanded.

[the remark this answers is not in the document]
Comment (jouni) [reply articles/ds/components/crud/index.asciidoc:273]: I’ll check what’s wrong with the preview build.

File: frontend/demo/component/crud/crud-basic.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
+
```
Comment (jouni) [inline frontend/demo/component/crud/crud-basic.ts:1]:
```suggestion
```

File: articles/ds/components/crud/index.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (magi42) [reply articles/ds/components/crud/index.asciidoc:273]: OK. Ah, preview doesn't show Java examples, hadn't noticed that before.

**Comment by jouni** [comment]
@tomivirkki, @yuriy-fix, @sissbruecker, @vursen, @web-padawan, anyone have a clue what to do about these Java issues? I would like to get this merged asap, but these feel blocking (the other things @anezthes mentions in the description seem minor):

> Setting `setAsRequiredEnabled` on the binder crashes the example. Used `setRequiredIndicatorVisible` instead but fields never become invalid. (All fields are required by default in Fusion.)

> Can't bind birthday to binder (`DatePicker` uses `LocalDate`, dataset uses `Date`, changing the dataset crashed the examples).

**Comment by sissbruecker** [comment]
> anyone have a clue what to do about these Java issues? 

I can take a look.

**Comment by sissbruecker** [comment]
Started working on this. Todo list:

- [x] Setting setAsRequiredEnabled on the binder crashes the example. Used setRequiredIndicatorVisible instead but fields never become invalid. (All fields are required by default in Fusion.)
- [x] Can't bind birthday to binder (DatePicker uses LocalDate, dataset uses Date, changing the dataset crashed the examples)
- [ ] `CrudOpenEditor.java` is broken, with a NPE:
```
Caused by: java.lang.NullPointerException: null
        at com.vaadin.flow.component.grid.Grid.ensureOwner(Grid.java:2908) ~[vaadin-grid-flow-21.0.0.alpha2.jar:na]
        at com.vaadin.flow.component.grid.GridColumnOrderHelper.setColumnOrder(GridColumnOrderHelper.java:74) ~[vaadin-grid-flow-21.0.0.alpha2.jar:na]
        at com.vaadin.flow.component.grid.Grid.setColumnOrder(Grid.java:4170) ~[vaadin-grid-flow-21.0.0.alpha2.jar:na]
        at com.vaadin.flow.component.grid.Grid.setColumnOrder(Grid.java:4139) ~[vaadin-grid-flow-21.0.0.alpha2.jar:na]
        at com.vaadin.demo.component.crud.CrudOpenEditor.setupGrid(CrudOpenEditor.java:89) ~[classes/:na]
        at com.vaadin.demo.component.crud.CrudOpenEditor.<init>(CrudOpenEditor.java:36) ~[classes/:na]
```

**Comment by sissbruecker** [comment]
@jouni Fixed the issues you listed above and also a broken example.

The setup in the demos is not ideal, but I guess a lot of that is due to missing APIs (e.g. no way to create an empty Crud and add columns explicitely from Java). I guess it's your call if it's good enough now 🙂.

**Pull request merged**